feat: centralize frontend custom types into types.ts (#48)

GauravNardia opened the pull request:
This PR resolves #48 by consolidating all custom frontend TypeScript types into a single file:
`web/src/lib/types.ts`

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Refactor**
  - Centralized and standardized type definitions across the application by importing shared types, reducing duplication and improving maintainability.
- **Chores**
  - Updated development scripts to remove an unused command-line option for improved script clarity.
- **New Features**
  - Expanded shared type definitions to cover more UI components and application state, enhancing type safety and consistency.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

geekyharsh05 commented:
@coderabbitai review

GauravNardia commented:
@geekyharsh05 Yeah! I've tested on my end. Everything is good

geekyharsh05 commented on apps/web/src/hooks/useFileOperations.ts:
```diff
@@ -1,12 +1,9 @@
+import type { DeleteFileParams } from "../lib/types";
```
Here too! Use alias-based imports

AbhiArya20 replied:
```suggestion
import type { DeleteFileParams } from "@/lib/types";

geekyharsh05 commented on apps/web/src/hooks/useRequest.ts:
```diff
@@ -1,19 +1,7 @@
+import type { UseRequestParams, UseRequestReturn } from "../lib/types";
```
same goes here

GauravNardia replied:
okay i fix this

AbhiArya20 commented on apps/web/src/hooks/useRequest.ts:
```diff
@@ -1,19 +1,7 @@
+import type { UseRequestParams, UseRequestReturn } from "../lib/types";
```
```suggestion
import type { UseRequestParams, UseRequestReturn } from "@/lib/types";

AbhiArya20 commented on apps/web/src/hooks/useUpload.ts:
```diff
@@ -1,25 +1,12 @@
 // TODO: Move to useFileOperations.ts
 
+import type { AxiosError, CreateFolderParams } from "../lib/types";
```
```suggestion
import type { AxiosError, CreateFolderParams } from "@/lib/types";

AbhiArya20 commented on apps/web/src/lib/types.ts:
```diff
@@ -1,9 +1,85 @@
+	onCreateFolder: (folderName: string, parentId?: string | undefined) => void;
+}
+
+export interface UploadFileDialogProps {
+	open: boolean;
+	onOpenChange: (open: boolean) => void;
+	onUpload: (files: FileList) => void;
+}
+
+export interface UploadFolderDialogProps {
+	open: boolean;
+	onOpenChange: (open: boolean) => void;
+	onUpload: (files: FileList) => void;
+}
```
These two types look very similar - consider consolidating them into a single shared type to reduce duplication.

GauravNardia replied:
sure!

logscore commented on apps/web/src/lib/types.ts:
```diff
@@ -1,9 +1,85 @@
+	refetch: () => void;
+};
+
+export type AxiosError = {
+	response?: {
+		data?: {
+			message?: string;
+		};
+	};
+	message: string;
+};
```
I second this. @GauravNardia you will need to adjust the error handling in `useUpload.ts` to this:

```
onError: (error: AxiosError) => {
	console.error("Error creating folder:", error);
	const errorMessage = error.message || "Failed to create folder";
	toast.error(errorMessage);
},
```

GauravNardia replied:
yes removed it

GauravNardia replied:
@logscore 
so I need to write it in types.ts again ?
`onError: (error: AxiosError) => {
	console.error("Error creating folder:", error);
	const errorMessage = error.message || "Failed to create folder";
	toast.error(errorMessage);
},

GauravNardia commented:
**All requested changes have been addressed:**

-  Replaced all `../../../` import paths with alias imports (e.g., `@/lib/types`)
-  Confirmed no local import paths to types are left
-  Verified the app runs without errors

Let me know if there’s anything else to tweak. Thanks!

logscore commented on apps/web/src/components/dialogs/upload-folder-dialog.tsx:
```diff
@@ -8,15 +8,10 @@ import {
-	onUpload: (files: FileList) => void;
-}
-
 export function UploadFolderDialog({ open, onOpenChange, onUpload }: UploadFolderDialogProps) {
```
```suggestion
export function UploadFolderDialog({ open, onOpenChange, onUpload }: UploadFileDialogProps) {
```

logscore commented on apps/web/src/components/dialogs/upload-folder-dialog.tsx:
```diff
@@ -8,15 +8,10 @@ import {
 } from "@/components/ui/dialog";
 import { UploadZone } from "@/components/upload/upload-zone";
 import { useEffect, useState, type FormEvent } from "react";
+import type { UploadFolderDialogProps } from "@/lib/types";
```
```suggestion
import type { UploadFileDialogProps } from "@/lib/types";
```

logscore reviewed (commented):
The imports need to be updated, then we can merge this in

GauravNardia commented on apps/web/src/hooks/useUpload.ts:
```diff
@@ -58,7 +45,7 @@ export function useUpload() {
 			toast.success("Folder created successfully");
 		},
-		onError: (error: AxiosError) => {
+		onError: (error: AxiosError<{ message?: string }>) => {
 			console.error("Error creating folder:", error);
 			const errorMessage = error.response?.data?.message || error.message || "Failed to create folder";
```
I've updated Imports @logscore

GauravNardia commented:
### ✅ Changes Made @logscore 

- Updated the dialog props and error handling based on feedback
- Removed redundant typings from `AxiosError` (already provides `message`)
- Fixed `UploadFolderDialogProps` reference to the correct `UploadFileDialogProps`
- Cleaned up and ensured consistency across the components

Let me know if there’s anything else I should refine. Thanks for the review!

logscore commented:
Looks good. Going to be working on merging a few PRs later today. This'll be added on to them
